ragtime.jdbc/src/ragtime/jdbc.clj

```diff
@@ -17,10 +17,18 @@
                         [[:id "varchar(255)"]
                          [:created_at "varchar(32)"]]))
 
-(defn- ensure-migrations-table-exists [db-spec migrations-table]
-  (try
-    (sql/execute! db-spec [(migrations-table-ddl migrations-table)])
-    (catch SQLException _)))
+(defn- table-exists?
+  [connection table-name]
+  (when connection
+    (let [results (.getTables (.getMetaData connection) nil nil table-name nil)
```

Comment:
Rather than using the Java methods directly here, let's instead use the [metadata-query](https://clojure.github.io/java.jdbc/#clojure.java.jdbc/metadata-query) function in clojure.java.jdbc.